classes/catkin.bbclass

```diff
@@ -6,17 +6,29 @@ ROS_USE_PYTHON3 ??= "no"
 
 inherit cmake ${@'distutils3-base' if bb.utils.to_boolean(d.getVar('ROS_USE_PYTHON3', True)) else 'distutils-base'} ros faulty-solibs
 
-DEPENDS_prepend = "${@'' if (d.getVar('BPN', True) == 'catkin') or (d.getVar('BPN', True) == 'catkin-runtime') else 'catkin-native '}"
+# Prepend build dependency on "catkin-runtime" and "catkin-native"
+# if the package is not "catkin" or "catkin-runtime"
+DEPENDS =+ "${@'' if d.getVar('BPN', True) in ('catkin', 'catkin-runtime') else 'catkin-runtime catkin-native'}"
+
+# Prepend run dependency on "catkin-runtime" for base packages
+# if the package is not "catkin" or "catkin-runtime"
+RDEPENDS_${PN} =+ "${@'' if d.getVar('BPN', True) in ('catkin', 'catkin-runtime') else 'catkin-runtime'}"
+
+# Prepend run dependency on "catkin" for dev packages
+# if the package is not "catkin" or "catkin-runtime"
+RDEPENDS_${PN}-dev =+ "${@'' if d.getVar('BPN', True) in ('catkin', 'catkin-runtime') else 'catkin'}"
```

Comment:
@dominiquehunziker please disregard this comment it's not relevant to meta-ros as such.